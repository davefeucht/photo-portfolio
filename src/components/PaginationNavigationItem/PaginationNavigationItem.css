.pagination-navigation__item {
    margin: 3px;
    color: #FFFFFF;
}

.pagination-navigation__item.selected {
    font-weight: bold;
    font-size: larger
}

.pagination-navigation__item.deselected {
    cursor: pointer;
}
