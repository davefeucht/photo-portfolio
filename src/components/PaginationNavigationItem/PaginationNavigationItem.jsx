/****************
* PaginationNavigationItem component displays one item of the pagination navigation
****************/

import React from "react";
import PropTypes from 'prop-types';
import './PaginationNavigationItem.css';

const PaginationNavigationItem = ({ content, selectedState, navigationFunction }) => {
    return (
        <div className={`pagination-navigation__item ${selectedState}`} onClick={navigationFunction ? () => navigationFunction(parseInt(content)) : undefined}>{content}</div>
    )
};

PaginationNavigationItem.displayName = 'PaginationNavigationItem';

PaginationNavigationItem.propTypes = {
    content: PropTypes.string.isRequired,
    selectedState: PropTypes.string.isRequired,
    navigationFunction: PropTypes.func
};

export default PaginationNavigationItem;
