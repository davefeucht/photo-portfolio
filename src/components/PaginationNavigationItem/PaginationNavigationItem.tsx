/** **************
* PaginationNavigationItem component displays one item of the pagination navigation
*************** */

import './PaginationNavigationItem.css';

import * as React from 'react';

interface PaginationNavigationItemProps {
    content: string,
    selectedState: string,
    navigationFunction(content: number): void,
    isLast: boolean
}

const PaginationNavigationItem: React.FC<PaginationNavigationItemProps> = ({
    content, selectedState, navigationFunction, isLast
}) => {
    return (
        <div
            className={`pagination-navigation__item ${selectedState}`}
            onClick={navigationFunction ? () => navigationFunction(parseInt(content)) : undefined}
        >
            {content}
            {!isLast ? ' â€¢ ' : ''}
        </div>
    );
};

PaginationNavigationItem.displayName = 'PaginationNavigationItem';

export default PaginationNavigationItem;
