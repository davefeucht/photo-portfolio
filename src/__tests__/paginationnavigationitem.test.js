import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import PaginationNavigationItem from '../components/PaginationNavigationItem/PaginationNavigationItem';

const content = '1';
const selectedState = 'selected';
const navigationFunction = () => {};
const isLast = false;

test('PaginationNavigationItem displays', async () => {
    let component;
    let tree;
    component = renderer.create(
        <MemoryRouter>
            <PaginationNavigationItem content={content} selectedState={selectedState} navigationFunction={navigationFunction} isLast={isLast} />
        </MemoryRouter>
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
