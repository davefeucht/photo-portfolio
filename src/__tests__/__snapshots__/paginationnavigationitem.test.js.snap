// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationNavigationItem displays 1`] = `
<div
  className="pagination-navigation__item selected"
  onClick={[Function]}
>
  1
   â€¢ 
</div>
`;
